File: www/adbrix.js
/**
 * Created by cuter74 on 2016. 9. 9..
 */

var exec = require('cordova/exec');

function Adbrix(){}

Adbrix.prototype.endSession = function(callback) {
    exec(callback,function(err){console.log("adbrix error!");},"Adbrix","end",[]);
};

Adbrix.prototype.menuName = function (menuName,callback) {
    exec(callback,function(err){console.log("adbrix menu error!");},"Adbrix","retention",[menuName]);
};
Adbrix.prototype.member = function (menuName,callback) {
    exec(callback,function(err){console.log("adbrix member error!");},"Adbrix","member",[]);
};
Adbrix.prototype.age = function (age,callback) {
    exec(callback,function(err){console.log("adbrix age error!");},"Adbrix","age",[age]);
};
Adbrix.prototype.gender = function (gender,callback) {
    exec(callback,function(err){console.log("adbrix gender error!");},"Adbrix","gender",[gender]);
};
var adbrix = new Adbrix();

module.exports = adbrix;

File: src/ios/Adbrix.m
//
//  Adbrix.m
//  DreamDNA
//
//  Created by mscbrain on 2016. 10. 6..
//
//
#import "Adbrix.h"
#import <Cordova/CDVPlugin.h>
#import <AdBrix/AdBrix.h>
#import <IgaworksCore/IgaworksCore.h>

@implementation Adbrix

-(void) retention :(CDVInvokedUrlCommand*)command
{
    NSString* menuName  = [command.arguments objectAtIndex:0];

    NSLog(@"menuName============!!==%@", menuName);
    [AdBrix retention:menuName];
    
}
-(void) age :(CDVInvokedUrlCommand*)command
{
    NSInteger* age  = [[command.arguments objectAtIndex:0] intValue];
    
 //   NSLog(@"menuName============!!==%@", [age stringValue]);
    [IgaworksCore setAge:age];
    
}
-(void) gender:(CDVInvokedUrlCommand *)command
{
    NSString* gender  = [command.arguments objectAtIndex:0];
    
    NSLog(@"menuName============!!==%@", gender);
    if([gender isEqualToString:@"M"]==YES)
        [IgaworksCore setGender:IgaworksCoreGenderMale];
    else
        [IgaworksCore setGender:IgaworksCoreGenderFemale];

    
}
@end
